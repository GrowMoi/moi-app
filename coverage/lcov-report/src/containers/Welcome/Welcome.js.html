<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/Welcome/Welcome.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/containers/Welcome</a> Welcome.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components/native';
import { Actions } from 'react-native-router-flux';
import { LinearGradient } from 'expo';
&nbsp;
class Welcome extends Component {
  goToHome = <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >Actions.home();</span>
&nbsp;
  render() {
    return (
      &lt;Gradient colors={['#5FE89F', '#5EDFA7', '#66CDF7']}
        start={[0.1, 0.1]}
        end={[1, 1]}&gt;
        &lt;Title color='white' onPress={this.goToHome}&gt;
          &lt;FormattedMessage id="commons.welcome" /&gt;!
        &lt;/Title&gt;
        &lt;SimpleText&gt;
          &lt;FormattedMessage id="commons.text" /&gt;
        &lt;/SimpleText&gt;
      &lt;/Gradient&gt;
    );
  }
}
&nbsp;
/**
 * Styled components that use in this Scene
 */
&nbsp;
const Title = styled.Text`
  color: ${({ color }) =&gt; color &amp;&amp; color};
  font-size: 25;
  padding-vertical: 10;
  font-weight: bold;
  background-color: transparent;
`;
&nbsp;
const SimpleText = styled.Text`
  background-color: transparent;
  font-weight: bold;
`;
&nbsp;
const Gradient = styled(LinearGradient)`
  flex: 1;
  align-self: stretch;
  align-items: center;
  justify-content: center;
`;
&nbsp;
export default Welcome;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 04 2017 08:15:06 GMT-0500 (-05)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
