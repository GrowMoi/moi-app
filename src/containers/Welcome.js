import React, { Component } from 'react';
import styled from 'styled-components/native';
import { LinearGradient } from 'expo';

class Welcome extends Component {
  render() {
    return (
      <Gradient colors={['#5FE89F', '#5EDFA7', '#66CDF7']}
        start={[0.1, 0.1]}
        end={[1, 1]}>
        <Title color='white'>Welcome to Moi app!</Title>
        <SimpleText>This is a first component</SimpleText>
      </Gradient>
    );
  }
}

/**
 * Styled components that use in this Scene
 */

const Title = styled.Text`
  color: ${({ color }) => color && color};
  font-size: 25;
  font-weight: bold;
  background-color: transparent;
`;

const SimpleText = styled.Text`
  background-color: transparent;
  font-weight: bold;
`;

const Gradient = styled(LinearGradient)`
  flex: 1;
  align-self: stretch;
  align-items: center;
  justify-content: center;
`;

export default Welcome;
