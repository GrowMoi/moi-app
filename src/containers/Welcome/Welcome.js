import React, { Component } from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components/native';
import { Actions } from 'react-native-router-flux';
import { LinearGradient } from 'expo';

class Welcome extends Component {
  goToHome = () => Actions.tabbars();

  render() {
    return (
      <Gradient colors={['#5FE89F', '#5EDFA7', '#66CDF7']}
        start={[0.1, 0.1]}
        end={[1, 1]}>
        <Title color='white' onPress={this.goToHome}>
          <FormattedMessage id="commons.welcome" />!
        </Title>
        <SimpleText>
          <FormattedMessage id="commons.text" />
        </SimpleText>
      </Gradient>
    );
  }
}

/**
 * Styled components that use in this Scene
 */

const Title = styled.Text`
  color: ${({ color }) => color && color};
  font-size: 25;
  padding-vertical: 10;
  font-weight: bold;
  background-color: transparent;
`;

const SimpleText = styled.Text`
  background-color: transparent;
  font-weight: bold;
`;

const Gradient = styled(LinearGradient)`
  flex: 1;
  align-self: stretch;
  align-items: center;
  justify-content: center;
`;

export default Welcome;
