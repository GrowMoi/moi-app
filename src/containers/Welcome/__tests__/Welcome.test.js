import React from 'react';
import createComponentWithIntl from '../../../commons/utils/intl-jest';
import Welcome from '../Welcome';

describe('Welcome Scene', () => {
  it('renders expected markup', () => {
    const props = {
      name: 'welcome',
    };

    const wrapper = createComponentWithIntl(<Welcome {...props}/>)
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
