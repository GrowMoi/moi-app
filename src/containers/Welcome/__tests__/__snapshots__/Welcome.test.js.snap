// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Scene renders expected markup 1`] = `
<ExponentLinearGradient
  colors={
    Array [
      4284475551,
      4284407719,
      4284927479,
    ]
  }
  end={
    Array [
      1,
      1,
    ]
  }
  locations={null}
  start={
    Array [
      0.1,
      0.1,
    ]
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    color="white"
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontFamily": "Museo900",
          "fontSize": 25,
          "fontWeight": "bold",
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Hello
    </Text>
    !
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "fontFamily": "FuturaBold",
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      This is a first component
    </Text>
  </Text>
</ExponentLinearGradient>
`;
